trie/trie.go

```diff
@@ -287,34 +380,46 @@ func concat(s1 []byte, s2 ...byte) []byte {
 	return r
 }
 
-func (t *Trie) resolve(n node) node {
+func (t *Trie) resolve(n node, prefix, suffix []byte) (node, error) {
 	if n, ok := n.(hashNode); ok {
-		return t.resolveHash(n)
+		return t.resolveHash(n, prefix, suffix)
+	}
+	return n, nil
+}
+
+type MissingNodeError struct {
+	RootHash, NodeHash   common.Hash
+	KeyPrefix, KeySuffix []byte
+}
+
+func (err *MissingNodeError) Error() string {
+	return fmt.Sprintf("Missing trie node %064x", err.NodeHash)
+}
+
+func unhandledError(err error) {
+	if glog.V(logger.Error) {
+		glog.Errorf("Unhandled trie error: %v", err)
```

Comment:
`Errorf` is not a method, unfortunately.
But you can combine the error and logging checks:

```go
if err := operation(); err != nil && glog.V(logger.Error) {
     glog.Errorf("Unhandled trie error: %v", err)
}
return nil
```